pl_bolts/models/rl/vanilla_policy_gradient_model.py

```diff
@@ -281,6 +209,49 @@ def calc_policy_loss(
         policy_loss = -log_prob_actions.mean()
         return log_prob, policy_loss
 
+    def train_batch(self) -> Tuple[List[torch.Tensor], List[torch.Tensor], List[torch.Tensor]]:
+        """
+        Contains the logic for generating a new batch of data to be passed to the DataLoader
+
+        Returns:
+            yields a tuple of Lists containing tensors for states, actions and rewards of the batch.
+        """
+        states = []
+        actions = []
+        scales = []
+
+        for _ in range(self.batch_size):
+
+            # take a step in the env
+            exp, reward, done = self.source.step(self.device)
+            self.env_steps += 1
+            self.total_steps += 1
+
+            # update the baseline
+            self.reward_sum += exp.reward
+            self.baseline = self.reward_sum / self.total_steps
+
+            # gather the experience data
+            states.append(exp.new_state)
```

Comment:
another question: can we already place the yield here? So that it says `yield new_state, action, scale` here instead of first collecting them and yielding in the end? This would probably save memory